applications/tari_base_node/src/builder.rs

```diff
@@ -473,19 +471,42 @@ where
         .set_base_node_public_key(base_node_public_key.clone())
         .await
         .expect("Problem setting local base node public key for transaction service.");
-    let mut oms_handle = wallet_handles
-        .get_handle::<OutputManagerHandle>()
-        .expect("OutputManagerService is not registered");
-    oms_handle
-        .set_base_node_public_key(base_node_public_key)
-        .await
-        .expect("Problem setting local base node public key for output manager service.");
     // Start the Output Manager UTXO Validation
-    oms_handle
-        .validate_utxos(UtxoValidationRetry::UntilSuccess)
-        .await
-        .expect("Problem starting the Output Manager Service Utxo Valdation process");
-
+    let state_machine = base_node_handles
+        .get_handle::<StateMachineHandle>()
+        .expect("Could not get State Machine handle");
+    let wallet_handles_clone = wallet_handles.clone();
+    task::spawn(async move {
+        let mut status_watch = state_machine.get_status_info_watch();
+        loop {
+            let bootstrapped = match status_watch.recv().await {
+                None => false,
+                Some(s) => match s {
+                    StatusInfo::Listening(li) => li.is_bootstrapped(),
+                    _ => false,
+                },
+            };
+            if bootstrapped {
+                let mut oms_handle = wallet_handles_clone
+                    .get_handle::<OutputManagerHandle>()
+                    .expect("OutputManagerService is not registered");
```

Comment:
I might be wrong, but the thread that is spawned here is never joined, so this panic will never be caught. The program will continue and this error will never be logged.

The main problem here is that a thread is getting started in the background, and there's no logical place for it to be joined. This makes me think it would be better to have the caller of this method be responsible for starting and joining this thread. It might also be better to listen for a state change event instead of polling the state each time

Reply:
We could possibly add another message to the log file if an exception occurred? Apart from that, this code is simple and works well under real base node and stress conditions. 

My proposal:

``` rust
            if bootstrapped {
                debug!(
                    target: LOG_TARGET,
                    "UTXO validation to start: Get OutputManagerHandle.",
                );
                let mut oms_handle = wallet_handles_clone
                    .get_handle::<OutputManagerHandle>()
                    .expect("OutputManagerService is not registered");
                debug!(
                    target: LOG_TARGET,
                    "UTXO validation to start: Set BaseNodePublicKey.",
                );
                oms_handle
                    .set_base_node_public_key(base_node_public_key)
                    .await
                    .expect("Problem setting local base node public key for output manager service.");
                debug!(
                    target: LOG_TARGET,
                    "UTXO validation to start: Attempting to start the Output Manager Service UTXO Valdation process.",
                );
                oms_handle
                    .validate_utxos(UtxoValidationRetry::UntilSuccess)
                    .await
                    .expect("Problem starting the Output Manager Service UTXO Valdation process");
                debug!(
                    target: LOG_TARGET,
                    "UTXO validation to start: UTXO Valdation process concluded successfully .",
                );
                break;
            }
```

Reply:
The expect messages will still not be logged. I would suggest that the getting of the handle happen before the spawn so that will panic and for the other two you can use a map error before the expect to log the problem e.g.
```Rust
oms_handle
                    .set_base_node_public_key(base_node_public_key)
                    .await
                    .map_err(|e| {
                            error!(target: LOG_TARGET, "An error occured: {}", e);
                            e
                     }).expect("Problem setting local base node public key for output manager service.");
```

Reply:
Done